### PR TITLE
Add error/validation when password is the same as before

### PR DESCRIPTION
**Pull request by BryanEliDimas**
Part of https://github.com/openstax/business-intel/issues/851

### CODE REVIEW

**Comment by BryanEliDimas** [comment]
Can I please get a code review? @steveburkett

File: spec/handlers/newflow/change_password_spec.rb [hunk 1]
```diff
@@ -27,19 +27,33 @@ module Newflow
...
-      }
-    end
+      before do
+        FactoryBot.create(:identity, user: user, password: 'password')
+      end
 
       it 'fails if password is too short' do
-        # create the password to begin with
-        FactoryBot.create(:identity, user: user, password: 'password')
+        params = {
```
Comment (steveburkett) [inline spec/handlers/newflow/change_password_spec.rb]: minor...this params construct (all thru the test) could be dryed up a bit thru a let(:params) and a let(:password) fields.

Comment (BryanEliDimas) [reply spec/handlers/newflow/change_password_spec.rb]: Nice, I'd like to do that but I can't use `let(:password)` from within an example 🤔 .

Comment (BryanEliDimas) [reply spec/handlers/newflow/change_password_spec.rb]: Done. I used nested `describe` blocks and `example do` blocks.

**Comment by BryanEliDimas** [comment]
@steveburkett Dry-ed up the specs as you suggested, it worked out well! Specs passed. Merging PR.

**Pull request merged**